de.lucaswerkmeister.corebird.json

```diff
@@ -0,0 +1,65 @@
+{
+  "app-id": "de.lucaswerkmeister.corebird",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.26",
```

Comment:
I’m currently running my development build of Corebird on Arch Linux with Gnome 3.30, and it seems to be working fine, but I haven’t tried out this manifest at all, since I couldn’t figure out how to build it. Is there some documentation on how to get from this JSON file to an installed Flatpak app?

Also, if I bump the GNOME version to 3.30 here, will people on other distributions (where the native GNOME is older) still be able to install this?

Sorry if these are noob questions, I’m very new to the world of Flatpak.

Reply:
No problema! :+1: ...in your case, you should be able to build it and run it running these:

```shell
$ cd whenever_your_flatpak_files_are_located_at/
$ flatpak-builder build --force-clean --install-deps-from=flathub --install --user de.lucaswerkmeister.corebird.json
# ...and to clean-up everything (the following will uninstall the artifact, its dependencies, and everything build-related
$ flatpak uninstall --user de.lucaswerkmeister.corebird
$ flatpak uninstall --unused --user
$ rm -Rf .flatpak-builder/ build/ ~/.var/app/de.lucaswerkmeister.corebird/
```

Reply:
Thank you :heart: I had to set up the `flathub` remote in the user context first (`flatpak remote-add --user --gpg-import=/var/lib/flatpak/repo/flathub.trustedkeys.gpg flathub https://flathub.org/repo/flathub.flatpakrepo`), but then this runs – until I get this error:

> Error: Can't load desktop file /home/lucas/git/flathub/.flatpak-builder/rofiles/rofiles-rhzHLJ/files/share/applications/de.lucaswerkmeister.corebird.desktop: Error opening file /home/lucas/git/flathub/.flatpak-builder/rofiles/rofiles-rhzHLJ/files/share/applications/de.lucaswerkmeister.corebird.desktop: No such file or directory

`.flatpak-builder/rofiles/rofiles-rhzHLJ/` seems to be empty after `flatpak-builder` terminates, but I’m guessing that its contents were moved to `build/`. In `build/files/share/applications/`, a desktop file exists, but it’s `org.baedert.corebird.desktop`, not `de.lucaswerkmeister.corebird.desktop`, because I haven’t updated most occurrences of the `org.baedert.corebird` identifier in the source code yet. Will Flatpak insist on the `de.lucaswerkmeister.corebird.desktop` name? In that case I can try to update the source code accordingly.

Reply:
More likely. To my knowledge, in those cases I believe using `"rename-desktop-file": "org.baedert.corebird.desktop"` will do it. Moreover, when you have time, you can change that and later on remove that line in the manifest :clap:

Reply:
Thanks. `rename-desktop-file` works for that, but then D-Bus is the next candidate with the `org.baedert` in it (not just in the file name, but also in the contents this time), and after that no doubt some others as well… my current status is that adjusting the D-Bus service file is enough to make the Flatpak build succeed, but then I can’t run it (`org.freedesktop.DBus.Error.ServiceUnknown`).

I think I’ll try to rename some more things in the fork (I should be able to test them for Flatpak by changing the manifest to read from my local git checkout), and put out a 1.7.4.2 release if I’m successful. Thanks a lot for your help so far!

Reply:
Done, including update to GNOME 3.30 :)